web/src/OverviewActionBar.tsx

```diff
@@ -326,6 +405,91 @@ export function FilterRadioButton(props: FilterRadioButtonProps) {
   )
 }
 
+function FilterTermField(props: { filterSet: FilterSet }) {
+  const globalTerm = props.filterSet.term
```

Comment:
It's referring to the term object on the global state, rather than the local value of the input field. I had to separate the two to implement debouncing, but it does create a bit of a mismatched state (like using the `invalid` that from the global term, while the local input value may be different).

On a somewhat related note, I don't think the `debounce` function that comes with material-ui is working like I think it should... let me dig into that first.

Reply:
I'm not sure about calling that state global, either.

Do you like any of these options?
1. change props to `props: { initialValue: string }` and change the caller to pass in filterSet.term.input (which also
2. just get rid of this line and pass `props.filterSet.term` to `useState` directly, avoiding the naming problem
3. rename `globalTerm` to `termFromUrl` (though that feels like then this function knows too much about what its caller is doing - but that's already a problem with the comment, but I think that'd let you eliminate the comment and the introduction of the term "global")

Reply:
Yes! It absolutely makes sense to change this to `initialTerm` and just pass that down as props. Thanks for that suggestion.